keps/sig-apps/20190318-Graduate-CronJob-to-Stable.md

```diff
@@ -0,0 +1,552 @@
+- [Proposal](#proposal)
+  - [Promote CronJob API to v1](#promote-cronjob-api-to-v1)
+  - [Rearchitect CronJob controller](#rearchitect-cronjob-controller)
+    - [Existing controller](#existing-controller)
+    - [Proposed rearchitect](#proposed-rearchitect)
+      - [Informers and Caches](#informers-and-caches)
+      - [Multiple workers](#multiple-workers)
+      - [Handling Cron aspect](#handling-cron-aspect)
+      - [Metrics](#metrics)
+  - [Add .status.lastSuccessfulTime](#add-statuslastsuccessfultime)
+  - [Add .status.nextScheduleTime](#add-statusnextscheduletime)
+  - [Add Counters](#add-counters)
+  - [Add .status.conditions](#add-statusconditions)
+  - [CatchUp concurrency support](#catchup-concurrency-support)
+  - [Use master timezone instead of UTC](#use-master-timezone-instead-of-utc)
+  - [Fix applicable open issues](#fix-applicable-open-issues)
+  - [Scale Targets for GA](#scale-targets-for-ga)
+    - [CronJob Limits](#cronjob-limits)
+    - [Frequency of launched jobs](#frequency-of-launched-jobs)
+- [V1 API](#v1-api)
+- [v1beta1 changes](#v1beta1-changes)
+- [Validations](#validations)
+- [Tests](#tests)
+  - [E2E test](#e2e-test)
+    - [Existing test cases](#existing-test-cases)
+    - [New test cases](#new-test-cases)
+  - [Conformance Tests](#conformance-tests)
+  - [Unit Tests](#unit-tests)
+- [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+
+[CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) is a Kubernetes API that creates Job object on a schedule specified by a cron spec. It is in beta status since v1.8. This document lays out the plan to promote it to stable.
+
+## Motivation
+
+CronJob definition has been stable for the last few releases and is useful to run periodic tasks in kubernetes cluster. This API adds the ability to add cron facility to a cluster. We feel the API is ready to be promoted to Stable and be supported longterm by the community.
+
+## Goals
+
+- Rearchitect CronJob controller
+- .status.lastSuccessfulTime
+- .status.nextRunTime
+- Use master timezone instead of UTC
+- Fix open issues and add tests
+
+## Non-Goals
+- Changing API field or meaning
+- [When creating a cronjob, Reference an existing job](https://github.com/kubernetes/kubernetes/issues/81329)
+- Add `CatchUp` concurrency policy
+
+## Proposal
+
+### Promote CronJob API to v1
+We will formally promote the GA and create `batch/v1/CronJob` in the [batch/v1 API](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/batch/v1/types.go). The older versions of CronJob API (batch/v1beta1, batch/v2alpha1) will be deprecated following the [deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/) 
+
+### Rearchitect CronJob controller
+
+#### Existing controller
+
+The current implementation of the CronJob controller is different that the other workload controllers. 
```

Comment:
Spell out explicitly that it does not follow https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/controllers.md

Reply:
added.